pkg/operator/metriccontroller/fsync_controller.go

```diff
@@ -81,12 +87,18 @@ func (c *FSyncController) sync(ctx context.Context, syncCtx factory.SyncContext)
 		return fmt.Errorf("unexpected type, expected Matrix, got %T", matrix)
 	}
 
+	degraded := false
```

Comment:
Consider: degradedMsg string to append values to.

Reply:
+ 1 yeah lets be very direct that we have observed performance thresholds which can endanger the stability of the cluster.